### ISSUE TITLE
SEI system tests intermittently failing

### ISSUE THREAD

**Issue by taylor-swanson** [elastic/integrations#6475]
As I've been working on pkg-spec and ECS updates, I've noticed the system tests from certain packages will intermittently fail. I would often have to re-run the tests 2-3 times before they eventually pass.

I noticed this initially in [this PR](https://github.com/elastic/integrations/pull/6326), but unfortunately the jenkins results have been rotated/deleted, so I'm not sure exactly what package was failing there.

The other package I've seen show this problem is `zoom`. Initially noticed in [this PR](https://github.com/elastic/integrations/pull/6328), but I also created a [standalone PR](https://github.com/elastic/integrations/pull/6458) where the problem happened there as well.

What triggered the failure for zoom was a lack of events appearing in the expected data stream. Here's an excerpt from the logs (lines omitted for brevity, [full log here](https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-6328/runs/6/steps/4051/log/?start=0)):

```
[2023-06-02T19:08:39.237Z]  Container elastic-package-service-zoom-webhook-https-1  Creating
[2023-06-02T19:08:39.237Z]  Container elastic-package-service-zoom-webhook-http-1  Creating
[2023-06-02T19:08:39.541Z]  Container elastic-package-service-zoom-webhook-http-1  Created
[2023-06-02T19:08:39.541Z]  Container elastic-package-service-zoom-webhook-https-1  Created
[2023-06-02T19:08:39.541Z]  Container elastic-package-service-zoom-webhook-https-1  Starting
[2023-06-02T19:08:39.541Z]  Container elastic-package-service-zoom-webhook-http-1  Starting
[2023-06-02T19:08:39.944Z]  Container elastic-package-service-zoom-webhook-http-1  Started
[2023-06-02T19:08:39.944Z]  Container elastic-package-service-zoom-webhook-https-1  Started
[2023-06-02T19:08:39.944Z] 2023/06/02 19:08:39 DEBUG running command: /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker/docker-compose.yml -p elastic-package-service ps -q
[2023-06-02T19:08:39.944Z] 2023/06/02 19:08:39 DEBUG Wait for healthy containers: 5304c81ef7fc55481191d065e47b547d7d8735424a8c86cc86b2104bca8bd883,f03b65980efe62c2490397b76517f1e1c38fc2b769b945bade0b81dba0fcfa52
[2023-06-02T19:08:39.944Z] 2023/06/02 19:08:39 DEBUG output command: /bin/docker inspect 5304c81ef7fc55481191d065e47b547d7d8735424a8c86cc86b2104bca8bd883 f03b65980efe62c2490397b76517f1e1c38fc2b769b945bade0b81dba0fcfa52
[2023-06-02T19:08:40.045Z] 2023/06/02 19:08:39 DEBUG Container status: {"Config":{"Image":"akroh/stream:v0.2.0","Labels":{"com.docker.compose.config-hash":"5e8a339d4636f6d5b1bd7cace20a91e5093ad5f6a69bfd4ca9a8d081db90c32e","com.docker.compose.container-number":"1","com.docker.compose.depends_on":"","com.docker.compose.image":"sha256:0c869fdd39ad06efbf3ce7693dc26cc2eae511cea43b090d1db509a3802117c1","com.docker.compose.oneoff":"False","com.docker.compose.project":"elastic-package-service","com.docker.compose.project.config_files":"/var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker/docker-compose.yml","com.docker.compose.project.working_dir":"/var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker","com.docker.compose.service":"zoom-webhook-http","com.docker.compose.version":"2.17.2"}},"ID":"5304c81ef7fc55481191d065e47b547d7d8735424a8c86cc86b2104bca8bd883","State":{"Status":"running","ExitCode":0,"Health":null}}
[2023-06-02T19:08:40.045Z] 2023/06/02 19:08:39 DEBUG Container status: {"Config":{"Image":"akroh/stream:v0.2.0","Labels":{"com.docker.compose.config-hash":"4109103f3b0a51f7ce537d61122f97f36636b381f8d47fd72c76552c551621e4","com.docker.compose.container-number":"1","com.docker.compose.depends_on":"","com.docker.compose.image":"sha256:0c869fdd39ad06efbf3ce7693dc26cc2eae511cea43b090d1db509a3802117c1","com.docker.compose.oneoff":"False","com.docker.compose.project":"elastic-package-service","com.docker.compose.project.config_files":"/var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker/docker-compose.yml","com.docker.compose.project.working_dir":"/var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker","com.docker.compose.service":"zoom-webhook-https","com.docker.compose.version":"2.17.2"}},"ID":"f03b65980efe62c2490397b76517f1e1c38fc2b769b945bade0b81dba0fcfa52","State":{"Status":"running","ExitCode":0,"Health":null}}
[2023-06-02T19:08:40.045Z] 2023/06/02 19:08:39 DEBUG run command: /bin/docker network connect elastic-package-stack_default elastic-package-service-zoom-webhook-http-1
[2023-06-02T19:08:40.145Z] 2023/06/02 19:08:40 DEBUG adding service container elastic-package-service-zoom-webhook-http-1 internal ports to context
[2023-06-02T19:08:40.146Z] 2023/06/02 19:08:40 DEBUG running command: /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker/docker-compose.yml -p elastic-package-service config
[2023-06-02T19:08:40.146Z] 2023/06/02 19:08:40 DEBUG creating test policy...

...

[2023-06-02T19:08:54.299Z] 2023/06/02 19:08:54 DEBUG running command: /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-6328/src/github.com/elastic/integrations/packages/zoom/_dev/deploy/docker/docker-compose.yml -p elastic-package-service kill -s SIGHUP zoom-webhook-http
[2023-06-02T19:08:54.399Z]  Container elastic-package-service-zoom-webhook-http-1  Killing
[2023-06-02T19:08:54.399Z]  Container elastic-package-service-zoom-webhook-http-1  Killed
[2023-06-02T19:08:54.399Z] 2023/06/02 19:08:54 DEBUG checking for expected data in data stream...
[2023-06-02T19:08:54.399Z] 2023/06/02 19:08:54 DEBUG found 0 hits in logs-zoom.webhook-ep data stream: index_not_found_exception: no such index [logs-zoom.webhook-ep] Status=404
[2023-06-02T19:08:55.404Z] 2023/06/02 19:08:55 DEBUG found 0 hits in logs-zoom.webhook-ep data stream: index_not_found_exception: no such index [logs-zoom.webhook-ep] Status=404
[Previous line repeats many times ...]
[2023-06-02T19:18:54.387Z] 2023/06/02 19:18:54 DEBUG reassigning original policy back to agent...

...

[2023-06-02T19:19:07.631Z]  Container elastic-package-service-zoom-webhook-http-1  Stopping
[2023-06-02T19:19:07.631Z]  Container elastic-package-service-zoom-webhook-http-1  Stopping
[2023-06-02T19:19:07.631Z]  Container elastic-package-service-zoom-webhook-https-1  Stopping
[2023-06-02T19:19:07.632Z]  Container elastic-package-service-zoom-webhook-https-1  Stopping
[2023-06-02T19:19:07.632Z]  Container elastic-package-service-zoom-webhook-http-1  Stopped
[2023-06-02T19:19:07.632Z]  Container elastic-package-service-zoom-webhook-http-1  Removing
[2023-06-02T19:19:07.632Z]  Container elastic-package-service-zoom-webhook-http-1  Removed
[2023-06-02T19:19:07.832Z]  Container elastic-package-service-zoom-webhook-https-1  Stopped
[2023-06-02T19:19:07.832Z]  Container elastic-package-service-zoom-webhook-https-1  Removing
[2023-06-02T19:19:07.832Z]  Container elastic-package-service-zoom-webhook-https-1  Removed
[2023-06-02T19:19:07.832Z]  Network elastic-package-service_default  Removing
[2023-06-02T19:19:07.933Z]  Network elastic-package-service_default  Removed
[2023-06-02T19:19:07.933Z] 2023/06/02 19:19:07 DEBUG deleting data in data stream...
[2023-06-02T19:19:07.933Z] Error: error running package system tests: could not complete test run: could not find hits in logs-zoom.webhook-ep data stream
script returned exit code 1
```

As of now, `zoom` is passing without issue, so the failures seem to have resolved on their own.

Update (2023-06-05): I have not updated the branch for [standalone PR](https://github.com/elastic/integrations/pull/6458) and the test is still failing. It seems it may have captured the error state, so if anyone is curious, feel free to look at that PR.

**Comment by andrewkroh** [comment]
The stream container is timing out before it can connect successfully to the agent webhook endpoint. `elastic-package` sends a SIGHUP to the `stream` container after the agent policy has been successfully applied. Then `stream` delays 5 seconds before trying 10 connections each separated by 1s.

So I would configure stream to do more [retries](https://github.com/elastic/stream/blob/e56c6e82b1f1931163e053b76b98f7a51a0a4aa9/pkg/output/options.go#L11-L13) and see if that solves the issue. This could be an indication of some kind of slow down within the Agent or Filebeat.

```
elastic-package-service-zoom-webhook-https-1  | {"level":"info","ts":"2023-06-02T21:22:04.672Z","caller":"command/root.go:108","msg":"Waiting for signal.","start-signal":"SIGHUP"}
elastic-package-service-zoom-webhook-http-1   | {"level":"info","ts":"2023-06-02T21:22:04.729Z","caller":"command/root.go:108","msg":"Waiting for signal.","start-signal":"SIGHUP"}
elastic-package-service-zoom-webhook-http-1   | {"level":"info","ts":"2023-06-02T21:22:21.748Z","caller":"command/root.go:119","msg":"Delaying connection.","delay":5}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:26.749Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:27.751Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:28.753Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:29.754Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:30.756Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:31.757Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:32.758Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:33.760Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:34.761Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | {"level":"debug","ts":"2023-06-02T21:22:35.763Z","caller":"output/util.go:27","msg":"Connecting...","address":"http://elastic-agent:9080/zoom"}
elastic-package-service-zoom-webhook-http-1   | Error: Head "http://elastic-agent:9080/zoom": dial tcp 172.18.0.7:9080: connect: connection refused
```

**Comment by andrewkroh** [comment]
Upon deeper inspection of the agent log file I see:

```
Unit state changed http_endpoint-default-http_endpoint-zoom-ff340252-e114-42ea-8451-5e23340ce6c9 (STARTING->FAILED): [failed to reload inputs: 1 error: Error creating runner from config: crc.secret is required when crc.provider is defined accessing config]
Unit state changed http_endpoint-default (STARTING->FAILED): [failed to reload inputs: 1 error: Error creating runner from config: crc.secret is required when crc.provider is defined accessing config]
```

**Comment by ebeahan** [comment]
I was able to reproduce locally using a branch before https://github.com/elastic/integrations/commit/4d9e74a28df87b57c01f0ac035b9e292525d2806 was merged against `8.9.0-SNAPSHOT`:

```
Killing elastic-package-service_zoom-webhook-https_1 ... done
2023/06/05 15:44:45 DEBUG checking for expected data in data stream...
2023/06/05 15:44:45 DEBUG found 0 hits in logs-zoom.webhook-ep data stream: index_not_found_exception: no such index [logs-zoom.webhook-ep] Status=404
2023/06/05 15:44:46 DEBUG found 0 hits in logs-zoom.webhook-ep data stream: index_not_found_exception: no such index [logs-zoom.webhook-ep] Status=404
```

Maybe https://github.com/elastic/integrations/pull/6411 resolved the system test failures for `zoom`?
